### ISSUE TITLE
omegaconf_ignore not working

### ISSUE THREAD

**Issue by PoojaAg18** [omry/omegaconf#1025]
```python
from dataclasses import dataclass, field
@dataclass
class HasIgnoreMetadata:
   normal_field: int = 1
   field_ignored: int = field(default=2, metadata={"omegaconf_ignore": True})

cfg = OmegaConf.create(HasIgnoreMetadata())
cfg
```

Result - {'normal_field': 1, 'field_ignored': 2}

omegaconf_ignore doesn't ignore the field

**Comment by PoojaAg18** [comment]
it seems #988 is not included in omegaconf 2.3 version

**Comment by Jasha10** [comment]
Hello,
OmegaConf version 2.3 has not been released yet. The current release is OmegaConf v2.2.3.
The `omegaconf_ignore` feature will be included in the v2.3 release.

You can try a dev release to preview OmegaConf v2.3:
`pip install omegaconf==2.3.0.dev1`
https://pypi.org/project/omegaconf/2.3.0.dev1/
